src/powershell/Public/ConvertTo-FinOpsSchema/ConvertTo-FinOpsSchema.ps1

```diff
@@ -0,0 +1,221 @@
+It currently understands how to convert Cost Management cost data using the latest schemas as of September 2023.
+Older schemas may not be fully supported. Please review output and report any issues to https://aka.ms/ftk.
+You can pipe objects to ConvertTo-FinOpsSchema from an exported or downloaded CSV file using Import-Csv or ConvertFrom-Csv and pipe to Export-Csv to save as a CSV file.
+
+.PARAMETER ActualCost
+Specifies the actual cost data to be converted. Object must be a supported Microsoft Cost Management schema.
+
+.PARAMETER AmortizedCost
+Specifies the amortized cost data to be converted. Object must be a supported Microsoft Cost Management schema.
+
+.PARAMETER Version
+FOCUS schema version to convert data to. Allowed values: 0.5, 1.0-workingdraft, 1.0-preview. Default = 0.5.
+
+.EXAMPLE
+ConvertTo-FinOpsSchema -ActualCost (Import-Csv my-actual-cost-details.csv) -AmortizedCost (Import-Csv my-amortized-cost-details.csv) | Export-Csv my-cost-details-in-focus.csv
+
+.NOTES
+Version: 
+Author: 
+Date: 
+#>
+
+function ConvertTo-FinOpsSchema {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory=$true)]
+        [object]$ActualCost,
+
+        [Parameter(Mandatory=$false)]
+        [object]$AmortizedCost,
+
+        [Parameter()]
+        [ValidateSet("0.5", "1.0-workingdraft", "1.0-preview")]
+        [string]$Version = "0.5"
+    )
+    
+    # Define the columns to add to the output
+    $columnsToAdd = @(
+        'ftk_AccountName', 
+        'ftk_AccountId', 
+        'ftk_AccountOwnerId', 
+        'ftk_AvailabilityZone', 
+        'ftk_EffectivePrice'
+        )
+    
+    # Combine ActualCost and AmortizedCost
+    $CombinedCost = $ActualCost + $AmortizedCost
+    Write-Host "Combined Cost Data Count: $($CombinedCost.Count)"
+    
+    # Perform transformations based on the version
+    switch ($Version) {
+        "0.5" {
+            if ($null -ne $CombinedCost -and $CombinedCost.Count -gt 0) {
+                $firstRow = $CombinedCost[0]
+                # If the first row of the data does not contain the billingAccountId or BillingAccountId column, throw an error
+                if (-not ($firstRow.PSObject.Properties.Name -contains 'billingAccountId' -or $firstRow.PSObject.Properties.Name -contains 'BillingAccountId')) {
+                    throw "Cost details schema not supported."
+                }
+            # else throw an error 
+            } else {
+                throw "Combined Cost data is empty or null."
+            }
+            # TODO: Filter the actual cost data to rows where ChargeType == "Purchase" and PricingModel == "Reservation" or "SavingsPlan" (leave amortized cost as-is).
+            # This to ensure that we only have rows with Purchase, reservation or savings plan data. Only applicable to actual cost data. Leaving the amortized cost data as-is.
+
+            # 1. Uppercase Column Names Modifications
+            $CombinedCost = $CombinedCost | ForEach-Object {
```

Comment:
Is this looping thru every row? I wonder if there's an easier/faster way to change the property names 🤔